### PR TITLE
add milvus vector store

### PR DESCRIPTION
**Pull request by wodka**
Create Milvus Vector Store to use in llamaindex

connection is based on https://github.com/milvus-io/milvus-sdk-node

- [x] connection to server
- [x] connection to collection
- [x] adding documents
- [x] deleting documents
- [x] querying documents
- [x] persisting documents
- [x] add example

### CODE REVIEW

File: examples/milvus/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- `OPENAI_API_KEY`: Your OpenAI key
+
+2. `cd` Into the `examples` directory
+3. run `npm i`
+
+## Load the data
+
+This sample loads the same dataset of movie reviews as sample dataset. You can install https://github.com/zilliztech/attu to inspect the loaded data.
+
+run `ts-node milvus/load`
```
Comment (marcusschiesser) [inline examples/milvus/README.md:28]: is that working without `ts-node` being installed globally?

**Review by marcusschiesser — COMMENTED** [review]
@wodka Looks great. Thanks. Just a few nitpicks in the comments. Is the PR good to merge from your side?

**Comment by marcusschiesser** [comment]
@wodka does your TS package work with NextJS? We often need to add exceptions for other libraries (see https://github.com/run-llama/LlamaIndexTS?tab=readme-ov-file#note-nextjs)

File: examples/milvus/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - Self Hosted https://milvus.io/docs/install_standalone-docker.md
+- An OpenAI API Key
+
+## Setup
+
+1. Set your env variables:
+
+- `MILVUS_ADDRESS`: Address of your Milvus Vector Store (like localhost:19530)
+- `MILVUS_USERNAME`: empty or username for your Milvus Vector Store
+- `MILVUS_PASSWORD`: empty or password for your Milvus Vector Store
```
[the remark this answers is not in the document]
Comment (wodka) [reply examples/milvus/README.md:18]: added

**Comment by wodka** [comment]
> @wodka does your TS package work with NextJS? We often need to add exceptions for other libraries (see https://github.com/run-llama/LlamaIndexTS?tab=readme-ov-file#note-nextjs)

I haven't tried, but when looking at the milvus-sdk-node package that it's using there are modification required for the the proto files. https://github.com/milvus-io/milvus-sdk-node/issues/170#issuecomment-1575295886

I did add them now to the readme @marcusschiesser 

is there an easy way for me to test that this is working?

File: examples/milvus/README.md (hunk 1 above)
Comment (wodka) [reply examples/milvus/README.md]: ^^ kind of, modified the command now to look like `npx ts-node milvus/load` -> then it uses the local one. (I also added this to the astradb readme)

File: README.md [hunk 3]
```diff
@@ -104,6 +104,42 @@ const nextConfig = {
 module.exports = nextConfig;
 ```
 
+### NextJS with Milvus:
```
Comment (marcusschiesser) [inline README.md:107]: @wodka, that's a bit big for the main README. Do we need to do this just for Next.JS with Milvus or for any NextJS project (after merging this PR)? If it's the former please move this README to the Milvus one

File: examples/astradb/README.md [hunk 4]
```diff
@@ -25,10 +25,10 @@ Here are two sample scripts which work well with the sample data in the Astra Po
 
 This sample loads the same dataset of movie reviews as the Astra Portal sample dataset. (Feel free to load the data in your the Astra Data Explorer to compare)
 
-run `ts-node astradb/load`
+run `npx ts-node astradb/load`
```
Comment (marcusschiesser) [inline examples/astradb/README.md:28]: @wodka thanks!

File: examples/milvus/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcusschiesser) [reply examples/milvus/README.md:18]: what's `MILVUS_SSL`?

[the remark this answers is not in the document]
Comment (wodka) [reply examples/milvus/README.md:18]: to enable SSL: https://github.com/milvus-io/milvus-sdk-node/blob/main/milvus/MilvusClient.ts#L47C4-L47C4

**Comment by marcusschiesser** [comment]
> > @wodka does your TS package work with NextJS? We often need to add exceptions for other libraries (see https://github.com/run-llama/LlamaIndexTS?tab=readme-ov-file#note-nextjs)
> 
> I haven't tried, but when looking at the milvus-sdk-node package that it's using there are modification required for the the proto files. [milvus-io/milvus-sdk-node#170 (comment)](https://github.com/milvus-io/milvus-sdk-node/issues/170#issuecomment-1575295886)
> 
> I did add them now to the readme @marcusschiesser
> 
> is there an easy way for me to test that this is working?

You would need a NextJS project (use `npx create-llama`), create a local build of LlamaIndex, and then link this build to the project using `npm link` - a bit cumbersome.  
Otherwise, please wait till we add this to our e2e tests (see https://github.com/run-llama/LlamaIndexTS/issues/276),

File: README.md (hunk 3 above)
Comment (wodka) [reply README.md:107]: will check this sometime tomorrow - thx for the hint on how to best test this

File: examples/milvus/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcusschiesser) [reply examples/milvus/README.md:18]: Sorry, i mean it's missing in the readme

File: README.md (hunk 3 above)
Comment (marcusschiesser) [reply README.md:107]: @wodka if you don't have time for this, I'll try it with the next release

**Comment by hiepxanh** [comment]
hope this will be merge soon 💌  I'm looking for this PR

**Comment by EmanuelCampos** [comment]
bump @wodka any updates for this PR?

**Comment by dmitriyzhuk** [comment]
Yeah, waiting for this one too 👍

**Comment by wodka** [comment]
I am still running it in my system - so the code itself works, but I did not yet have time to investigate in combination with nextjs - maybe next week

**Comment by marcusschiesser** [comment]
Thanks a lot @wodka - we added it with https://github.com/run-llama/LlamaIndexTS/pull/640

**Pull request closed without merging**